search/ranker.cpp

```diff
@@ -608,4 +648,58 @@ void Ranker::ProcessSuggestions(vector<RankerResult> & vec) const
     ++i;
   }
 }
+
+string Ranker::GetLocalizedRegionInfoForResult(RankerResult const & result) const
+{
+  size_t const kMaxNumParts = 2;
+
+  auto const type = result.GetBestType(&m_params.m_preferredTypes);
+
+  storage::TCountryId id;
+  if (!result.GetCountryId(m_infoGetter, type, id))
+    return {};
+
+  vector<string> parts;
+  GetPathToRoot(id, m_countries, [&](storage::TCountryId const & id) {
+    parts.push_back(GetLocalizedCountryName(id));
+  });
+
+  if (parts.size() > kMaxNumParts)
+    parts.erase(parts.begin(), parts.end() - kMaxNumParts);
+
+  my::EraseIf(parts, [&](string const & s) { return s.empty(); });
+
+  if (!parts.empty())
+    return strings::JoinStrings(parts, ", ");
+
+  // Tries to get at least localized name for |id|, if |id| is a
+  // discussed territory.
+  auto name = GetLocalizedCountryName(id);
+  if (!name.empty())
+    return name;
+
+  // Tries to transform map name to the full name.
+  name = id;
+  storage::CountryInfo::FileName2FullName(name);
+  if (!name.empty())
+    return name;
+
+  return {};
+}
+
+string Ranker::GetLocalizedCountryName(storage::TCountryId const & id) const
```

Comment:
Maybe it's worth writing tests on methods like that?

Reply:
Probably :)

Reply:
Done